spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/source/SparkColumnarReaderFactory.java

```diff
@@ -28,10 +29,12 @@
 
 class SparkColumnarReaderFactory implements PartitionReaderFactory {
   private final int batchSize;
+  private final ParquetReaderType parquetReaderType;
 
-  SparkColumnarReaderFactory(int batchSize) {
+  SparkColumnarReaderFactory(int batchSize, ParquetReaderType readType) {
```

Comment:
This one will also get a reference to `BatchReadConf` as opposed to separate variables. The batch size validation can be moved to `BatchReadConf` itself.

Reply:
Fixed. Thanks